apps/dashboard/src/components/conditions-editor/variable-select.tsx

```diff
@@ -0,0 +1,265 @@
+  options: Array<{ label: string; value: string }>;
+  onSelect: (value: string) => void;
+  selectedValue?: string;
+};
+
+const VariablesList = React.forwardRef<HTMLInputElement, VariablesListProps>(
+  ({ options, onSelect, selectedValue }, ref) => {
+    const [hoveredOptionIndex, setHoveredOptionIndex] = useState(-1);
+    const listRef = useRef<HTMLUListElement>(null);
+
+    const scrollToOption = (index: number) => {
+      if (!listRef.current) return;
+
+      const listElement = listRef.current;
+      const optionElement = listElement.children[index] as HTMLLIElement;
+
+      if (optionElement) {
+        const containerHeight = listElement.clientHeight;
+        const optionTop = optionElement.offsetTop;
+        const optionHeight = optionElement.clientHeight;
+
+        if (optionTop < listElement.scrollTop) {
+          // Scroll up if option is above visible area
+          listElement.scrollTop = optionTop;
+        } else if (optionTop + optionHeight > listElement.scrollTop + containerHeight) {
+          // Scroll down if option is below visible area
+          listElement.scrollTop = optionTop + optionHeight - containerHeight;
+        }
+      }
+    };
+
+    const next = () => {
+      if (hoveredOptionIndex === -1) {
+        setHoveredOptionIndex(0);
+        scrollToOption(0);
+      } else {
+        setHoveredOptionIndex((oldIndex) => {
+          const newIndex = oldIndex === options.length - 1 ? 0 : oldIndex + 1;
+          scrollToOption(newIndex);
+          return newIndex;
+        });
+      }
+    };
+
+    const prev = () => {
+      if (hoveredOptionIndex === -1) {
+        setHoveredOptionIndex(options.length - 1);
+        scrollToOption(options.length - 1);
+      } else {
+        setHoveredOptionIndex((oldIndex) => {
+          const newIndex = oldIndex === 0 ? options.length - 1 : oldIndex - 1;
+          scrollToOption(newIndex);
+          return newIndex;
+        });
+      }
+    };
+
+    return (
+      <div className="flex flex-col">
+        <input
+          className="sr-only"
+          ref={ref}
+          onFocusCapture={() => {
+            setHoveredOptionIndex(0);
+            scrollToOption(0);
+          }}
+          onKeyDown={(e) => {
+            if (e.key === 'ArrowDown') {
+              next();
+              e.preventDefault();
+            } else if (e.key === 'ArrowUp') {
+              prev();
+              e.preventDefault();
+            } else if (e.key === 'Enter') {
+              if (hoveredOptionIndex !== -1) {
+                onSelect(options[hoveredOptionIndex].value ?? '');
+                setHoveredOptionIndex(-1);
+              }
+            }
+          }}
+        />
+        <header className="flex items-center justify-between gap-1 border-b border-neutral-100 bg-neutral-50 p-1">
+          <span className="text-foreground-400 text-paragraph-2xs uppercase">Variables</span>
+          <KeyboardItem>{`{`}</KeyboardItem>
+        </header>
```

Comment:
to me the title is a bit misleading, as I'm sure that those are actually variables, more like fields or similar. wdyt?

Reply:
good catch, better name it `Fields`